src/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -896,5 +966,153 @@ public static void InvalidDateTimeStyles()
             DateTimeStyles style = DateTimeStyles.AssumeLocal | DateTimeStyles.AssumeUniversal;
             AssertExtensions.Throws<ArgumentException>("style", () => DateTime.ParseExact(strDateTime, formats, provider, style));
         }
+
+        public static IEnumerable<object[]> Parse_ValidInput_Suceeds_MemberData()
+        {
+            yield return new object[] { "1234 12", null, new DateTime(1234, 12, 1, 0, 0, 0) };
+            yield return new object[] { "12 1234", null, new DateTime(1234, 12, 1, 0, 0, 0) };
+            yield return new object[] { "12 1234 11", null, new DateTime(1234, 12, 11, 0, 0, 0) };
+            yield return new object[] { "1234 12 13", null, new DateTime(1234, 12, 13, 0, 0, 0) };
+            yield return new object[] { "12 13 1234", null, new DateTime(1234, 12, 13, 0, 0, 0) };
+            yield return new object[] { "1 1 1", null, new DateTime(2001, 1, 1, 0, 0, 0) };
+            yield return new object[] { "2 2 2Z", null, null };
+            yield return new object[] { "#10/10/2095#\0", null, new DateTime(2095, 10, 10, 0, 0, 0) };
+
+            DateTime today = DateTime.Today;
+
+            var hebrewCulture = new CultureInfo("he-IL");
+            hebrewCulture.DateTimeFormat.Calendar = new HebrewCalendar();
+            yield return new object[] { today.ToString(hebrewCulture), hebrewCulture, today };
+
+            var mongolianCulture = new CultureInfo("mn-MN");
+            yield return new object[] { today.ToString(mongolianCulture.DateTimeFormat.FullDateTimePattern, mongolianCulture), mongolianCulture, today };
+        }
+
+        [Theory]
+        [MemberData(nameof(Parse_ValidInput_Suceeds_MemberData))]
+        public static void Parse_ValidInput_Suceeds(string input, CultureInfo culture, DateTime? expected)
+        {
+            DateTime result = DateTime.Parse(input, culture);
+            if (expected != null)
```

Comment:
>if (expected != null) [](start = 12, length = 21)

you will not need this check after setting last param in the data

Reply:
Ok, I'll try TimeZoneInfo. Thanks.